diff --git mythplugins/mythgallery/mythgallery/iconview.cpp mythplugins/mythgallery/mythgallery/iconview.cpp
index da7f35b..d23f559 100644
--- mythplugins/mythgallery/mythgallery/iconview.cpp
+++ mythplugins/mythgallery/mythgallery/iconview.cpp
@@ -612,7 +619,7 @@ bool IconView::HandleMediaEscape(MediaMonitor *mon)
 {
     bool handled = false;
     QDir curdir(m_currDir);
-    QList<MythMediaDevice*> removables = mon->GetMedias(MEDIATYPE_DATA);
+    QList<MythMediaDevice*> removables = mon->GetMedias(MEDIATYPE_DATA|MEDIATYPE_MGALLERY);
     QList<MythMediaDevice*>::iterator it = removables.begin();
     for (; !handled && (it != removables.end()); it++)
     {
