Index: mythtv/configure
===================================================================
--- mythtv/configure	(revision 27299)
+++ mythtv/configure	(working copy)
@@ -2584,7 +2584,7 @@
 if enabled proc_opt_old; then
 # first try to tune based on processor info
 # Intel i7 processors (EMT64T capable)
-   if expr "$processor" : ".*Intel(R).*Core(TM) i7 CPU.*" > /dev/null ; then
+   if expr "$processor" : ".*Intel(R).*Core(TM) i[357] CPU.*" > /dev/null ; then
        cpu="core2"
 # EM64T era Intel Xeon
    elif expr "$processor" : ".*Intel(R) Xeon(R) CPU.*51[1-6][08]" > /dev/null ; then
