Index: mythtv/libs/libmyth/audiooutputpulse.cpp
===================================================================
--- mythtv/libs/libmyth/audiooutputpulse.cpp	(revision 27299)
+++ mythtv/libs/libmyth/audiooutputpulse.cpp	(working copy)
@@ -226,9 +226,10 @@
     QString fn_log_tag = "WriteAudio, ";
     pa_stream_state_t sstate = pa_stream_get_state(pstream);
 
-    // Do not write anything to pulse server if we are in pause mode
+#if 0    // Do not write anything to pulse server if we are in pause mode
     if (IsPaused())
         return;
+#endif
 
     VBAUDIOTS(fn_log_tag + QString("writing %1 bytes").arg(size));
 
