diff --git a/mythtv/bindings/python/Makefile b/mythtv/bindings/python/Makefile
index 0db784f..26a9c1a 100644
--- mythtv/bindings/python/Makefile
+++ mythtv/bindings/python/Makefile
@@ -5,11 +5,11 @@ ifdef INSTALL_ROOT
 endif
 
 REAL_PREFIX = $(PREFIX)
-ifneq ($(PREFIX:/=), /usr)
+#ifneq ($(PREFIX:/=), /usr)
 	ifneq ($(PREFIX:/=), /usr/local)
 		PREFIX_FLAGS=--prefix="$(PREFIX)"
 	endif
-endif
+#endif
 
 all: python_build
 
