diff --git mythtv/libs/libmyth/mythmediamonitor.cpp mythtv/libs/libmyth/mythmediamonitor.cpp
index 3dac2ff..8901064 100644
--- mythtv/libs/libmyth/mythmediamonitor.cpp
+++ mythtv/libs/libmyth/mythmediamonitor.cpp
@@ -811,12 +818,20 @@ QString MediaMonitor::defaultDevice(QString dbSetting,
     return device;
 }
 
+#ifdef Q_OS_WIN32
+# define DEFAULT_CD "e:"
+# define DEFAULT_DVD "e:"
+#else
+# define DEFAULT_CD "/dev/cdrom"
+# define DEFAULT_DVD "/dev/dvd"
+#endif
+
 /**
  * CDDevice, user-selected drive, or /dev/cdrom
  */
 QString MediaMonitor::defaultCDdevice()
 {
-    return defaultDevice("CDDevice", tr("Select a CD drive"), "/dev/cdrom");
+    return defaultDevice("CDDevice", tr("Select a CD drive"), DEFAULT_CD);
 }
 
 /**
@@ -825,7 +840,7 @@ QString MediaMonitor::defaultCDdevice()
 QString MediaMonitor::defaultVCDdevice()
 {
     return defaultDevice("VCDDeviceLocation",
-                         tr("Select a VCD drive"), "/dev/cdrom");
+                         tr("Select a VCD drive"), DEFAULT_CD);
 }
 
 /**
@@ -834,7 +849,7 @@ QString MediaMonitor::defaultVCDdevice()
 QString MediaMonitor::defaultDVDdevice()
 {
     return defaultDevice("DVDDeviceLocation",
-                         tr("Select a DVD drive"), "/dev/dvd");
+                         tr("Select a DVD drive"), DEFAULT_DVD);
 }
 
 /**
@@ -843,7 +858,7 @@ QString MediaMonitor::defaultDVDdevice()
 QString MediaMonitor::defaultCDWriter()
 {
     return defaultDevice("CDWriterDeviceLocation",
-                         tr("Select a CD writer"), "/dev/cdrom");
+                         tr("Select a CD writer"), DEFAULT_CD);
 }
 
 /**
@@ -855,7 +870,7 @@ QString MediaMonitor::defaultCDWriter()
 QString MediaMonitor::defaultDVDWriter()
 {
     QString device = defaultDevice("MythArchiveDVDLocation",
-                                   tr("Select a DVD writer"), "/dev/dvd");
+                                   tr("Select a DVD writer"), DEFAULT_DVD);
 
     return device;
 }
